### ISSUE TITLE
botan 2.19.0 osx build failure

### ISSUE THREAD

**Issue by chenrui333** [randombit/botan#2880]
👋 trying to build the latest release, but run into some build issue. The error log is as below:

<details>
<summary>build error</summary>

```
  build/include/botan/internal/sha3_round.h:35:27: error: deduced return types are a C++14 extension
  __attribute__((noinline)) decltype(auto) xor_CNs(const uint64_t A[25])
                            ^
  clang++ -fPIC -fvisibility=hidden -fstack-protector -pthread -stdlib=libc++ -std=c++11 -D_REENTRANT  -O3 -DBOTAN_IS_BEING_BUILT -Wall -Wextra -Wpedantic -Wshadow -Wstrict-aliasing -Wstrict-overflow=5 -Wcast-align -Wmissing-declarations -Wpointer-arith -Wcast-qual  -Ibuild/include -Ibuild/include/external -c src/lib/hash/sm3/sm3.cpp -o build/obj/lib/hash_sm3.o
  build/include/botan/internal/sha3_round.h:37:11: error: no viable conversion from returned value of type 'std::tuple<uint64_t, uint64_t, uint64_t, uint64_t, uint64_t>' (aka 'tuple<unsigned long long, unsigned long long, unsigned long long, unsigned long long, unsigned long long>') to function return type 'int'
     return std::tuple<uint64_t, uint64_t, uint64_t, uint64_t, uint64_t>(
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

full build log, https://github.com/Homebrew/homebrew-core/runs/4880276920
relates to Homebrew/homebrew-core#93448

**Comment by randombit** [comment]
Fixed in #2881 I will release a patch version later today

**Comment by chenrui333** [comment]
thank you!

**Comment by chenrui333** [comment]
I have tested out the patch, works fine for me.

**Comment by randombit** [comment]
Fix is now available in 2.19.1 release as well
